src/Compiler/Checking/MethodCalls.fs

```diff
@@ -1521,21 +1527,8 @@ let AdjustCallerArgForOptional tcVal tcFieldInit eCallerMemberName (infoReader:
                         // This should be unreachable but the error will be reported elsewhere
                         callerArgExpr
                 else
-                    if isNullableTy g calledArgTy  then 
```

Comment:
Quite a lot of logic removed here.  Does `AdjustCallerArgExpr` really cover all this?  Is it just a simplification or a fix?

Reply:
All this logic being so close to optional adjustments originates from FS-1075 being restricted to optional arguments at some point. Near completion it was expanded to cover all arguments but this code was never really moved to cover all arguments, which resulted in issues like #12994. 

Given `AdjustCallerArgExpr` runs for any argument after all fine-grained adjustments (which then unconditionally runs `AdjustExprForTypeDirectedConversions`) I chose to move it all to be colocated with the other TDC logic, just like I've done in that commit for the type checks in `AdjustRequiredTypeForTypeDirectedConversions`. 

The relevant commit is https://github.com/dotnet/fsharp/pull/13673/commits/da5d49b9a593d824782502f416c4b66c02e20bb5 and contains all changes pertaining to this.